### PR TITLE
Add support for GitHub Checks

### PR DESCRIPTION
**Pull request by ajvpot**
Adds support for a GitHub actions style check for vuln scan results in addition to or instead of a comment.
Gated behind a setting
Needs UI
Needs improvement to create check when we get the first webhook. For now we immediately create and complete it.

image
image

### CODE REVIEW

**Comment by ajvpot** [comment]
image
image

File: lunatrace/bsl/backend/src/github/actions/pr-comment-generator.ts [hunk 1]
```diff
@@ -67,7 +71,137 @@ function generatePullRequestCommentFromReport(projectId: string, scan: InsertedS
...
+    await hasura.UpdateBuildExistingReviewId({ id: buildId, existing_github_review_id });
+    return;
+  }
+
+  // Otherwise just update the existing review on the PR.  Very similar to above but we update instead
+  const githubReviewResponse = await github.UpdatePrReview({
+    pull_request_review_id: previousReviewId,
+    body,
+  });
+  const existing_github_review_id = githubReviewResponse.updatePullRequestReview?.pullRequestReview?.id;
```
Comment (freeqaz) [inline lunatrace/bsl/backend/src/github/actions/pr-comment-generator.ts:117]: Nit: camelCase for variables in JS

Comment (freeqaz) [reply lunatrace/bsl/backend/src/github/actions/pr-comment-generator.ts:117]: (we should make the linter do this)

Comment (ajvpot) [reply lunatrace/bsl/backend/src/github/actions/pr-comment-generator.ts:117]: I did this so I could inline it below, external api expects it this way.

**Pull request merged**